### PR TITLE
build: fetch `urcrypt` instead of vendoring it

### PR DESCRIPTION
**Pull request by matthew-levan**
As [`urcrypt`](https://github.com/urbit/urcrypt) now has its own repository, we can stop vendoring it and simply fetch it like any of our other third-party dependencies.

### CODE REVIEW

**Review by joemfb — DISMISSED** [review]
this will do

**Comment by joemfb** [comment]
@matthew-levan I seem to recall there being some question about the urcrypt header pattern here; has that been resolved? if so, feel free to merge.

**Comment by matthew-levan** [comment]
> @matthew-levan I seem to recall there being some question about the urcrypt header pattern here; has that been resolved? if so, feel free to merge.

Yea there were, but I never got around to resolving them. Looks like the build is working though in CI-- should we just merge and punt on the header particulars?

**Comment by joemfb** [comment]
This fails on my machine:

```
Makefile.am: installing 'build-aux/depcomp'
Makefile.am:23: error: 'pkgconfig_DATA' is used but 'pkgconfigdir' is undefined
autoreconf: error: automake failed with exit status: 1
```

**Comment by matthew-levan** [comment]
Ok, I don't get that error on my Mac. Which system are you using? Do you
have `/usr/local/include/urcrypt.h` installed? Which command produced that
stack trace? Thanks.

**Comment by matthew-levan** [comment]
Ok, thanks. Which version of macOS and which hardware?

**Comment by joemfb** [comment]
MacOS 12.6.1. 2021 Macbook Pro (m1 max)

**Comment by matthew-levan** [comment]
Maybe you have to install `pkg-config`. Do you have
`/usr/local/lib/pkgconfig` or `/usr/lib/pkgconfig`?

**Comment by joemfb** [comment]
After installing pkg-config via homebrew, i get a new error:

```
libtool: compile:  /usr/bin/clang -DHAVE_CONFIG_H -I. -I/private/var/tmp/_bazel_joemfb/f46182e3eb1bf4d7eff53357bd650651/sandbox/darwin-sandbox/3/execroot/__main__/external/aes_siv -I/private/var/tmp/_bazel_joemfb/f46182e3eb1bf4d7eff53357bd650651/sandbox/darwin-sandbox/3/execroot/__main__/bazel-out/darwin_arm64-opt/bin/external/aes_siv -I/private/var/tmp/_bazel_joemfb/f46182e3eb1bf4d7eff53357bd650651/sandbox/darwin-sandbox/3/execroot/__main__/bazel-out/darwin_arm64-opt/bin/external/openssl/openssl/include -I/private/var/tmp/_bazel_joemfb/f46182e3eb1bf4d7eff53357bd650651/sandbox/darwin-sandbox/3/execroot/__main__/bazel-out/darwin_arm64-opt/bin/external/urcrypt/urcrypt.ext_build_deps/openssl/include -I/private/var/tmp/_bazel_joemfb/f46182e3eb1bf4d7eff53357bd650651/sandbox/darwin-sandbox/3/execroot/__main__/bazel-out/darwin_arm64-opt/bin/external/urcrypt/urcrypt.ext_build_deps/secp256k1/include -march=native -std=c11 -Wextra -Wpedantic -Wall -Wall -g -O3 -MT keccak-tiny/libkeccak_tiny_la-keccak-tiny.lo -MD -MP -MF keccak-tiny/.deps/libkeccak_tiny_la-keccak-tiny.Tpo -c keccak-tiny/keccak-tiny.c -o keccak-tiny/libkeccak_tiny_la-keccak-tiny.o
clang: error: the clang compiler does not support '-march=native'
make[1]: *** [keccak-tiny/libkeccak_tiny_la-keccak-tiny.lo] Error 1
```

**Comment by matthew-levan** [comment]
Strange. Will try to look later. Are you planning to upgrade your Mac?
Perhaps that might fix it.

**Comment by joemfb** [comment]
This now works on my machine, but is broken on linux in CI.

Also, can you update INSTALL.md to mention the new pkg-config dependency for macos?

**Comment by matthew-levan** [comment]
Ok I fixed the build, and it should work with compilers that don't support `-march=native` also.

**Comment by joemfb** [comment]
After this update, the build failed with

```
error: possibly undefined macro: AC_MSG_NOTICE
```

After i installed autoconf-archive from homebrew, it builds correctly.

**Review by joemfb — CHANGES_REQUESTED** [review]

File: INSTALL.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ This will take a few minutes.
 
 ### macOS
 
-After installing `automake` and `libtool`, you're ready to build Vere.
+After installing `automake`, `autoconf-archive`, and `libtool`, you're ready to build Vere.
```
[the remark this answers is not in the document]
Comment (matthew-levan) [reply INSTALL.md:53]: tyty

**Review by joemfb — APPROVED** [review]
this will do

**Pull request merged**